File: PullToRefresh/app/src/main/java/com/example/pulltorefresh/MyAdapter.java
package com.example.pulltorefresh;

import android.content.Context;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import java.util.ArrayList;

public class MyAdapter extends ArrayAdapter<number> {
    Context context;
    int resource ;
    ArrayList<number> list;
    //SwipeRefreshLayout srl;
    static int i=0;

    public MyAdapter(Context context, int resource , ArrayList<number> list){
        super(context, resource ,list);
        this.context=context;
        this.resource=resource;
        this.list=list;
       // this.srl=srl;
    }

    @Nullable
    @Override
    public number getItem(int position) {
        return list.get(position);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
       convertView= LayoutInflater.from(context).inflate(R.layout.card_view_layout,parent,false);
       try{
           number x=getItem(position);

           TextView t=convertView.findViewById(R.id.text);
           t.setText((CharSequence) x);
       }catch (Exception e){
           System.out.println(e);
       }

      /* srl.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
           @Override
           public void onRefresh() {
               srl.setRefreshing(true);

               refresh(position);
           }
       });*/
       return convertView;
    }
   /* private void refresh(int position) {

        srl.setRefreshing(false);

        *//*new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                list.add(i);
                i++;
                MyAdapter.this.notifyDataSetChanged();
                srl.setRefreshing(false);
            }
        },3000);*//*
    }*/
}

File: ListView/app/src/main/java/com/example/listview/My_Adapter.kt
package com.example.listview

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.mailbox_layout.view.*

//for string array type

@Suppress("ClassName")
class My_Adapter(context: Context, resources:Int, var list: Array<String>):
    ArrayAdapter<String>(context,resources,list){


    override fun getItem(position: Int): String {
       return list.get(position)
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView=LayoutInflater.from(context).inflate(R.layout.mailbox_layout,parent,false)

        var t=convertView.t1
        var i=convertView.i1

        t.text=getItem(position)

        convertView.setOnClickListener(View.OnClickListener {
            Toast.makeText(context, "item "+(position+1), Toast.LENGTH_SHORT).show()
        })

        return convertView
    }

}